src/findersync.cs

```diff
@@ -54,6 +72,21 @@ interface FIFinderSyncProtocol
 
 		[Export ("toolbarItemToolTip")]
 		string ToolbarItemToolTip { get; }
+
+		[Export ("supportedMessageInterfaceNamesForItemWithURL:")]
+		string[] GetSupportedMessageInterfaceNames (NSUrl itemURL);
+
+		[Mac (10,13, onlyOn64 : true)]
+		[Export ("protocolForMessageInterface:")]
+		Protocol GetProtocol (NSFileProviderMessageInterface messageInterface);
+
+		[Mac (10,13, onlyOn64 : true)]
+		[Export ("exportedObjectForMessageInterface:itemURL:error:")]
+		NSObject GetExportedObject (NSFileProviderMessageInterface messageInterface, NSUrl itemUrl, [NullAllowed] out NSError error);
+
+		[Mac (10,13, onlyOn64 : true)]
+		[Export ("valuesForAttributes:forItemWithURL:completion:")]
+		void GetValues (string[] attributes, NSUrl itemUrl, Action<NSDictionary<NSString, NSObject>, NSError> completion);
 	}	
```

Comment:
Action too complex - we can guess the usage
Better use a custom delegate type in such case, it's self documenting